### PR TITLE
[stripe-v3] Update Stripe interface for new APIs

### PR DESCRIPTION
**Pull request by vgaudard**
* Add methods for new APIs:
    - PaymentMethod
    - Checkout
    - PaymentIntent

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

(changing an existing definition)
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: [stripe-js reference](https://stripe.com/docs/stripe-js/reference),
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by arjunkomath** [comment]
👍

**Comment by znagatomo** [comment]
👍

File: types/stripe-v3/index.d.ts [hunk 1]
```diff
@@ -27,6 +27,53 @@ declare namespace stripe {
         createSource(options: SourceOptions): Promise<SourceResponse>;
         retrieveSource(options: RetrieveSourceOptions): Promise<SourceResponse>;
         paymentRequest(options: paymentRequest.StripePaymentRequestOptions): paymentRequest.StripePaymentRequest;
+        createPaymentMethod(type: 'card' | 'card_present', element: elements.Element, data?: StripePaymentMethod): Promise<StripePaymentMethodResponse>;
```
Comment (marlosin) [inline types/stripe-v3/index.d.ts]: you can use the type `paymentMethodType` for `card` parameter

**Review by marlosin — CHANGES_REQUESTED** [review]
it'd be great some basic tests :+1:

**Comment by vgaudard** [comment]
Currently, tests in `types/stripe-v3` seem to use stripe-v2 (See [this](https://github.com/vgaudard/DefinitelyTyped/blob/d0d41ad9519a856f5792694eb8fd204f843869fd/types/stripe-v3/stripe-v3-tests.ts#L1)). Is there a reason for that or is that an oversight?

**Comment by marlosin** [comment]
> Currently, tests in `types/stripe-v3` seem to use stripe-v2 (See [this](https://github.com/vgaudard/DefinitelyTyped/blob/d0d41ad9519a856f5792694eb8fd204f843869fd/types/stripe-v3/stripe-v3-tests.ts#L1)). Is there a reason for that or is that an oversight?

TBH I don't know why it's like that, I'd guess who created v3 copied those tests.

**Review by marlosin — APPROVED** [review]
LGTM

**Pull request merged**